core/src/main/java/org/apache/spark/shuffle/sort/ShuffleExternalSorter.java

```diff
@@ -133,6 +144,26 @@
     this.peakMemoryUsedBytes = getMemoryUsage();
     this.diskWriteBufferSize =
         (int) (long) conf.get(package$.MODULE$.SHUFFLE_DISK_WRITE_BUFFER_SIZE());
+    this.checksumEnabled = (boolean) conf.get(package$.MODULE$.SHUFFLE_CHECKSUM());
+    if (this.checksumEnabled) {
+      this.partitionChecksums = new Adler32[numPartitions];
```

Comment:
this seems to be duplicated in a bunch of places can we put in a common place so that if we want to change or extend it would be easier?

Reply:
+1, I think I had a similar comment in the other WIP PR as well.

Reply:
Added `ShuffleChecksumHelper`